src/auto_pickup.cpp

```diff
@@ -517,45 +517,24 @@ void user_interface::show()
-                    // NOLINTNEXTLINE(cata-use-named-point-constants)
-                    fold_and_print( w_help, point( 1, 1 ), 999, c_white,
-                                    _(
-                                        "* is used as a Wildcard.  A few Examples:\n"
-                                        "\n"
-                                        "wooden arrow    matches the itemname exactly\n"
-                                        "wooden ar*      matches items beginning with wood ar\n"
-                                        "*rrow           matches items ending with rrow\n"
-                                        "*avy fle*fi*arrow     multiple * are allowed\n"
-                                        "heAVY*woOD*arrOW      case insensitive search\n"
-                                        "\n"
-                                        "Pickup based on item materials:\n"
-                                        "m:kevlar        matches items made of Kevlar\n"
-                                        "M:copper        matches items made purely of copper\n"
-                                        "M:steel,iron    multiple materials allowed (OR search)" )
-                                  );
-
-                    draw_border( w_help );
-                    wnoutrefresh( w_help );
-                } );
-                const std::string r = string_input_popup()
-                                      .title( _( "Pickup Rule:" ) )
-                                      .width( 30 )
-                                      .text( cur_rules[iLine].sRule )
-                                      .query_string();
+                string_input_popup_imgui popup( 60, cur_rules[iLine].sRule );
+                popup.set_label( _( "Pickup Rule:" ) );
+                std::string description = _(
+                                              "* is used as a Wildcard.  A few Examples:\n"
+                                              "\n"
+                                              "wooden arrow    matches the itemname exactly\n"
+                                              "wooden ar*      matches items beginning with wood ar\n"
+                                              "*rrow           matches items ending with rrow\n"
+                                              "*avy fle*fi*arrow     multiple * are allowed\n"
+                                              "heAVY*woOD*arrOW      case insensitive search\n"
+                                              "\n"
+                                              "Pickup based on item materials:\n"
+                                              "m:kevlar        matches items made of Kevlar\n"
+                                              "M:copper        matches items made purely of copper\n"
+                                              "M:steel,iron    multiple materials allowed (OR search)"
+                                          );
```

Comment:
For anything with a long description I would much rather see them use a custom ImGui window that can use both mono and variable‐width text, columns, colors, and so on.

Reply:
I'd love a flexible solution too, but I don't want to do that in this PR (and honestly don't plan to do it any time soon after it either).